packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -1696,12 +1700,59 @@ class BuildOwner {
   }
 }
 
-/// Elements are the instantiations of Widget configurations.
+/// An instantiation of a [Widget] at a particular location in the tree.
+///
+/// Widgets describe how to configure a subtree but the same widget can be used
+/// to configure multiple subtrees simultaneously because widget are immutable.
+/// An [Element] represents the use of a widget to configure a specific location
+/// in the tree. Over time, the widget associated with a given element can
+/// change, for example, if the parent widget rebuilds and creates a new widget
+/// for this location.
+///
+/// Elements form a tree. Most elements have a unique child, but some widgets
+/// (e.g., subclasses of [RenderObjectElement]) can have multiple children.
+///
+/// Elements have the following lifecycle:
 ///
-/// Elements can, in principle, have children. Only subclasses of
-/// RenderObjectElement are allowed to have more than one child.
+///  * The framework creates an element by calling [Widget.createElement].
+///  * The framework calls [mount] to add the newly created element to the tree
+///    at a given slot in a given parent. The [mount] method is responsible for
+///    inflating any child elements and calling [attachRenderObject] as
+///    necessary to attach any associated render objects to the render tree.
+///  * At this point, the element is considered "active" and might appear on
+///    screen.
+///  * At some point, the parent might decide to change the widget used to
+///    configure this element, for example because it rebuilt with new state.
+///    When this happens, the framework will call [update] with the new widget.
+///    The new widget will always have the [runtimeType] of the old widget. If
+///    the parent wishes to change the [runtimeType] of the widget at this
+///    location in the tree, can do so by unmounting this element and inflating
+///    a new element at this location.
+///  * At some point, an ancestor might decide to remove this element (or an
+///    intermediate ancestor) from the tree, which the ancestor does by calling
+///    [deactivateChild] on itself. Deactivating the intermediate ancestor will
+///    remove that element's render object from the render tree and add this
+///    element to the list of inactive elements, causing the framework to call
+///    [deactivate] on this element.
+///  * At this point, the element is considered "inactive" and will not appear
+///    on screen. An element can remain in the inactive state for at most one
+///    animation frame. At the end of the animation frame, any elements that are
```

Comment:
"at most one animation frame" sounds like it can survive a frame in some way (from one frame to the next is "one frame"). Maybe "until at the latest the end of the current frame"?